ada_feeding_perception/ada_feeding_perception/food_on_fork_categorical_naive_bayes_training.py

```diff
@@ -0,0 +1,288 @@
+    if not use_entire_dataset:
+        print("X_test", X_test.shape)
+        y_pred_test = clf.predict(X_test)
+        y_pred_test_proba = clf.predict_proba(X_test)
+        print(y_pred_test_proba)
+
+    # Get the train accuracy
+    acc_train = accuracy_score(y_train, y_pred_train)
+    print("Train accuracy", acc_train)
+    print("Train confusion matrix\n", confusion_matrix(y_train, y_pred_train))
+
+    # Get the test accuracy
+    if not use_entire_dataset:
+        acc_test = accuracy_score(y_test, y_pred_test)
+        print("Test accuracy", acc_test)
+        print("Test confusion matrix\n", confusion_matrix(y_test, y_pred_test))
+
+        # # Visualize what was learnt
+        # print("clf.feature_log_prob_", [m.shape for m in clf.feature_log_prob_])
+        # print("clf.feature_log_prob_", [np.sum(np.e**m, axis=1) for m in clf.feature_log_prob_])
+        # print("clf.feature_log_prob_", clf.feature_log_prob_, len(clf.feature_log_prob_), clf.feature_log_prob_[-1].shape)
+        # print("clf.class_count_", clf.class_count_, len(clf.class_count_), clf.class_count_[0].shape)
+
+        conditional_probabilities = [np.e**m for m in clf.feature_log_prob_]
+        prob_that_pixel_is_in_range_given_fof = []
+        prob_that_pixel_is_in_range_given_no_fof = []
+        for i in range(len(conditional_probabilities)):
+            try:
+                prob_that_pixel_is_in_range_given_fof.append(
+                    conditional_probabilities[i][1, 1]
+                )
+            except (
+                IndexError
+            ):  # If column 1 doesn't exist, that means the pixel was never in range
+                prob_that_pixel_is_in_range_given_fof.append(0)
+            try:
+                prob_that_pixel_is_in_range_given_no_fof.append(
+                    conditional_probabilities[i][0, 1]
+                )
+            except (
+                IndexError
+            ):  # If column 1 doesn't exist, that means the pixel was never in range
+                prob_that_pixel_is_in_range_given_no_fof.append(0)
+        prob_that_pixel_is_in_range_given_fof = np.array(
+            prob_that_pixel_is_in_range_given_fof
+        ).reshape(shape)
+        prob_that_pixel_is_in_range_given_no_fof = np.array(
+            prob_that_pixel_is_in_range_given_no_fof
+        ).reshape(shape)
+        plt_show_depth_img(
+            prob_that_pixel_is_in_range_given_no_fof,
+            title="Probability that a pixel is in range given no food on the fork",
+        )
+        plt_show_depth_img(
+            prob_that_pixel_is_in_range_given_fof,
+            title="Probability that a pixel is in range given food on the fork",
+        )
+        plt_show_depth_img(
+            prob_that_pixel_is_in_range_given_fof
+            - prob_that_pixel_is_in_range_given_no_fof,
+            title="P(px in range | FoF) - P(px in range | no FoF)",
+        )
+
+    # save model
+    if use_entire_dataset:
```

Comment:
It seems strange that whether or not to save the model is governed by the `use_entire_dataset` arg, as opposed to its own arg.

That is okay, but must be documented (e.g., in the docstrings of this file.)